### PR TITLE
Some elaboration of how to read compiler errors.

### PR DESCRIPTION
**Pull request by gigamonkey**
I get really frustrated when students don't even *try* to read the error messages. So here's some stuff to encourage them to do so and give them some pointers on how to.

### CODE REVIEW

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst [hunk 1]
```diff
@@ -285,9 +285,61 @@ Let's practice debugging some code!
 
 |CodingEx| **Coding Exercise: Compile Time Error 1**
```
Comment (bhoffman0) [inline _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst:286]: This got left behind. I think it goes above the exercise on line 343.

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst [hunk 2]
```diff
@@ -285,9 +285,61 @@ Let's practice debugging some code!
...
 
-Click on the |runbutton| button below to try and run the following code.  Look for an error message after the code.  This is called a **compile time error** because it is an error detected by the compiler.
+Reading error messages
```
Comment (bhoffman0) [inline _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst:288]:
```suggestion
Reading Error Messages
```

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst [hunk 3]
```diff
@@ -285,9 +285,61 @@ Let's practice debugging some code!
...
+----------------------
+
+In the exercise below you should click on the |runbutton| button below to
```
Comment (bhoffman0) [inline _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst:291]:
```suggestion
In the exercise below, you should click on the |runbutton| button to
```

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst [hunk 4]
```diff
@@ -285,9 +285,61 @@ Let's practice debugging some code!
...
+there's a colon (``:``) followed by a number. That number tells you the line
+number in the file where the compiler detected the error, in this case
+line 5.
+
+Error messages aren't always 100% accurate about where the error actually is;
+sometimes you actually need to change something a bit earlier in the program
+and sometimes a bit later. But the line number is the best place to start
+looking.
+
+After the line number and another colon is the actual error message. These
```
Comment (bhoffman0) [inline _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst:320]:
```suggestion
After the line number and another colon, you will find the actual error message. These
```

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst [hunk 5]
```diff
@@ -285,9 +285,61 @@ Let's practice debugging some code!
...
+quotation mark (``”``) before “Hi”. So it’s complaining about something being
+unclosed and it’s pointing us at a quotation mark. Usually quotation marks
+come in pairs called the open quote and the close quote, right? And if you
+look at the line of code you’ll see that there’s no closing quotation mark.
+Could that be the problem? Try adding a quotation mark and see if that fixes
+it!
+
+It is worth getting in the habit of really reading error messages. The people
+who wrote the Java compiler put in lot of work to try to make the error
+messages useful. If you actually read them you’ll soon learn to recognize
```
Comment (bhoffman0) [inline _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst:340]:
```suggestion
messages useful. If you read them, you’ll soon learn to recognize
```

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst [hunk 6]
```diff
@@ -365,7 +417,18 @@ Click on the |runbutton| button below to try and run the following code.  Look f
...
 
-Click on the |runbutton| button below to try and run the following code.  What is wrong this time?  Can you fix it?  After you fix the first error, you may encounter a 2nd error! Fix that one too! Hints: How do you end a statement in Java? Also, check for capitalization.
+Click on the |runbutton| button below to try and run the following code. What is
+wrong this time? The first error message will probably point you directly at the
+problem. But after you fix that you’ll probably get a `new` error! That happens
+sometimes because the compiler can’t even detect the second error until the
+first one is fixed.
+
+This is another reason it’s important to read the error message; if you just
+start trying to fix things without actually reading the error message you might
```
Comment (bhoffman0) [inline _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst:427]: Maybe too snarky? :-)

**Review by bhoffman0 — APPROVED** [review]
Good idea to add this section.

Also, Barb recently added some experimental micro-parsons problems at the end of this lesson that we could move into the text. The first 3 could go at the end of section 1.2.1 and the other 2 at the end of 1.2.2 Print methods,

File: _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst (hunk 6 above)
Comment (gigamonkey) [reply _sources/Unit1-Getting-Started/topic-1-2-java-intro.rst]: Is it better now?

**Pull request merged**